src/check/parse/Parser.zig

```diff
@@ -517,11 +503,16 @@ pub fn parseStmt(self: *Parser) ?IR.NodeStore.StatementIdx {
                     .ident_tok = start,
                     .region = .{ .start = start, .end = start },
                 } });
-                const statement_idx = self.store.addStatement(.{ .decl = .{
-                    .pattern = patt_idx,
-                    .body = idx,
-                    .region = .{ .start = start, .end = self.pos },
-                } });
+                const statement_idx = self.store.addStatement(.{
+                    .decl = .{
+                        .pattern = patt_idx,
+                        .body = idx,
+                        .region = .{
+                            .start = start,
+                            .end = self.pos -| 1, // we want the end of the previous token
```

Comment:
Does this need to be saturated? I would guess that it would be a bug if this saturates to 0. So likely we would actually prefer just `- 1` and to have it crash (thus enable the fuzzer to find bugs with regions).